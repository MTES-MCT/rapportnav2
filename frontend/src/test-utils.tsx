import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
