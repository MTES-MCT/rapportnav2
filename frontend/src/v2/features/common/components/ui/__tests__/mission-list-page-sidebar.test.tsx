import { Icon } from '@mtes-mct/monitor-ui'
import { render } from '../../../../../../test-utils'
import MissionListPageSidebar from '../mission-list-page-sidebar'

describe('MissionListPageSidebar', () => {
  it('should match the snapshot', () => {
    const wrapper = render(
      <MissionListPageSidebar
        items={[
          {
            key: 'mission',
            url: 'mission',
            icon: Icon.MissionAction
          }
        ]}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
