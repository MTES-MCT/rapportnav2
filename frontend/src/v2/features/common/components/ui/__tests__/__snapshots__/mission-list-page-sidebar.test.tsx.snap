// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MissionListPageSidebar > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <aside
        class="rs-sidebar"
        style="flex: 0 0px; width: 64px;"
      >
        <div
          class="sc-Pgsbw Qkkkg"
        >
          <div
            class="sc-gGKoUb hFDGXN"
            role="menu"
          >
            <button
              class="sc-kcLKEh sc-dFVmKS erhAT hjxgPz Element-IconButton sc-fbQrwq fiQlNT"
              role="menuitem"
              title="mission"
              type="button"
            >
              <span
                class="sc-hjsuWn eKbGln Element-IconBox"
              >
                <svg
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M214,2V0h-2V2h-4V0h-2V2h-3V18h14V2Zm1,14H205V4h10Z"
                    fill="currentColor"
                    transform="translate(-200)"
                  />
                  <rect
                    fill="currentColor"
                    height="1.5"
                    transform="translate(7 6.5)"
                    width="6"
                  />
                  <rect
                    fill="currentColor"
                    height="1.5"
                    transform="translate(7 9.5)"
                    width="6"
                  />
                  <path
                    d="M200,0h20V20H200Z"
                    fill="none"
                    transform="translate(-200)"
                  />
                  <path
                    d="M200,0h20V20H200Z"
                    fill="none"
                    transform="translate(-200)"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </aside>
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <aside
      class="rs-sidebar"
      style="flex: 0 0px; width: 64px;"
    >
      <div
        class="sc-Pgsbw Qkkkg"
      >
        <div
          class="sc-gGKoUb hFDGXN"
          role="menu"
        >
          <button
            class="sc-kcLKEh sc-dFVmKS erhAT hjxgPz Element-IconButton sc-fbQrwq fiQlNT"
            role="menuitem"
            title="mission"
            type="button"
          >
            <span
              class="sc-hjsuWn eKbGln Element-IconBox"
            >
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M214,2V0h-2V2h-4V0h-2V2h-3V18h14V2Zm1,14H205V4h10Z"
                  fill="currentColor"
                  transform="translate(-200)"
                />
                <rect
                  fill="currentColor"
                  height="1.5"
                  transform="translate(7 6.5)"
                  width="6"
                />
                <rect
                  fill="currentColor"
                  height="1.5"
                  transform="translate(7 9.5)"
                  width="6"
                />
                <path
                  d="M200,0h20V20H200Z"
                  fill="none"
                  transform="translate(-200)"
                />
                <path
                  d="M200,0h20V20H200Z"
                  fill="none"
                  transform="translate(-200)"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </aside>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
